code/datums/components/gas_connector.dm

```diff
@@ -0,0 +1,72 @@
+/datum/component/gas_connector
+	//Reference to the spawned pipe
+	var/obj/machinery/atmospherics/components/unary/internal_connector
+
+/datum/component/gas_connector/Initialize(obj/parent_obj, internal_volume)
+	. = ..()
+	internal_connector = new /obj/machinery/atmospherics/components/unary(parent_obj.loc)
```

Comment:
will this always look correct?

Reply:
The pipe will always be under the machine layer, so unless there is any weird stuff with the machine layering itself, it should look correct